### PR TITLE
Fixes issue blocking multiple sites.

### LINKED ISSUE #3368: Reader: Error and broken UI when blocking more than one site.
h/t @rachelsquirrel for the report.

To reproduce: Go to the reader post list and block a blog, then block a different blog in the list.  

The post list UI breaks, either going blank, or having very large blank areas when scrolling. An error is triggered in `WPTableViewHandler controllerDidChangeContent:` `[self.tableView endUpdates]` yielding the following message in the console:

> 2015-03-05 16:08:40.382 WordPress[4714:102869] CoreData: error: Serious application error.  An exception was caught from the delegate of NSFetchedResultsController during a call to -controllerDidChangeContent:.  Invalid update: invalid number of rows in section 0.  The number of rows contained in an existing section after the update (19) must be equal to the number of rows contained in that section before the update (20), plus or minus the number of rows inserted or deleted from that section (1 inserted, 1 deleted) and plus or minus the number of rows moved into or out of that section (0 moved in, 0 moved out). with userInfo (null)

### PR DESCRIPTION
**Pull request by aerych**
Closes #3368

### CODE REVIEW

**Comment by aerych** [comment]
@astralbodies would you be game for a review?

**Comment by astralbodies** [comment]
Reviewing now!

File: WordPress/Classes/ViewRelated/Reader/ReaderPostsViewController.m [hunk 1]
```diff
@@ -857,8 +872,8 @@ - (NSPredicate *)predicateForFetchRequest
 {
     NSPredicate *predicate;
 
-    if (self.postIDThatInitiatedBlock) {
-        predicate = [NSPredicate predicateWithFormat:@"topic = %@ AND (isSiteBlocked = NO OR postID = %@)", self.readerTopic, self.postIDThatInitiatedBlock];
+    if ([self.postIDsForUndoBlockCells count]) {
+        predicate = [NSPredicate predicateWithFormat:@"topic = %@ AND (isSiteBlocked = NO OR postID IN %@)", self.readerTopic, self.postIDsForUndoBlockCells];
```
Comment (astralbodies) [inline WordPress/Classes/ViewRelated/Reader/ReaderPostsViewController.m]: I remember when Core Data didn't support IN clauses with array objects. :stabby:

**Comment by astralbodies** [comment]
I think this looks good @aerych - I'm not super familiar with the Reader but this change looks like the right move. :shipit:

**Comment by astralbodies** [comment]
Actually I thought of one question - what happens in the rare case there are multiple posts from the same site in the Reader?

**Comment by aerych** [comment]
> what happens in the rare case there are multiple posts from the same site in the Reader

Its not so rare of a case. :)  All the posts belonging to the blocked site are [flagged as blocked](https://github.com/wordpress-mobile/WordPress-iOS/blob/develop/WordPress/Classes/Services/ReaderPostService.m#L410) in core data and excluded from fetch requests. To show the undo option, we specifically request the IDs of posts the user interacted with to initiate a block.  The blocked posts are purged the next time the reader's topic is synced.

**Comment by aerych** [comment]
Thanks @astralbodies !

**Pull request merged**